components/Modal/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AriaModal from 'react-aria-modal';
+import classNames from 'classnames';
+
+import styles from './index.scss';
+
+import Button from '../Button';
+
+export default class Modal extends React.Component {
+
+  static propTypes = {
+    actionText: PropTypes.string,
+    cancelText: PropTypes.string,
+    children: PropTypes.element.isRequired,
+    closeText: PropTypes.string,
+    onAction: PropTypes.func, // This should be a promise, so that onExit can be executed on success (nicer animation)
+    onExit: PropTypes.func.isRequired,
+    open: PropTypes.bool.isRequired,
+    titleText: PropTypes.string.isRequired
+  }
+
+  static defaultProps = {
+    actionText: '',
+    cancelText: '',
+    closeText: 'Close',
+    onAction: () => {},
+    verticallyCenter: true
+  };
+
+  state = {
+    modalHasEntered: false
+  };
+
+  onAction = ev => {
+    this.props.onAction(ev)
+      .then(() => this.onExit());
+  }
+
+  onExit = ev => {
+    // First set state for nicer animation
+    this.setState({
+      modalHasEntered: false
+    }, () => {
+      setTimeout(() => {
+        // Call parent onExit
+        this.props.onExit(ev);
```

Comment:
The onExit should be defined in the defaultProps

Reply:
onExit is required, since it should close the modal

Reply:
ah ok 🌮